src/main/java/com/minecrafttas/tasmod/networking/TASmodBufferBuilder.java

```diff
@@ -0,0 +1,212 @@
+package com.minecrafttas.tasmod.networking;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import com.minecrafttas.common.server.ByteBufferBuilder;
+import com.minecrafttas.common.server.interfaces.PacketID;
+import com.minecrafttas.tasmod.playback.PlaybackController.TASstate;
+import com.minecrafttas.tasmod.tickratechanger.TickrateChangerServer.TickratePauseState;
+
+import net.minecraft.nbt.CompressedStreamTools;
+import net.minecraft.nbt.NBTSizeTracker;
+import net.minecraft.nbt.NBTTagCompound;
+
+public class TASmodBufferBuilder extends ByteBufferBuilder{
+
+	public TASmodBufferBuilder(int id) {
+		super(id);
+	}
+	
+	public TASmodBufferBuilder(PacketID packet) {
+		super(packet);
+	}
+	
+	public TASmodBufferBuilder(ByteBuffer buf) {
+		super(buf);
+	}
+	
+	public TASmodBufferBuilder writeTASState(TASstate state) {
+		this.writeShort((short)state.ordinal());
+		return this;
+	}
+	
+	public TASmodBufferBuilder writeNBTTagCompound(NBTTagCompound compound) {
+		DataOutput out = new DataOutput() {
+			
+			@Override
+			public void writeUTF(String s) throws IOException {
+				writeString(s);
+			}
+			
+			@Override
+			public void writeShort(int v) throws IOException {
+				buffer.putShort((short) v);
+			}
+			
+			@Override
+			public void writeLong(long v) throws IOException {
+				buffer.putLong(v);
+			}
+			
+			@Override
+			public void writeInt(int v) throws IOException {
+				buffer.putInt(v);
+			}
+			
+			@Override
+			public void writeFloat(float v) throws IOException {
+				buffer.putFloat(v);
+			}
+			
+			@Override
+			public void writeDouble(double v) throws IOException {
+				buffer.putDouble(v);
+			}
+			
+			@Override
+			public void writeChars(String s) throws IOException {
+				writeString(s);
+			}
+			
+			@Override
+			public void writeChar(int v) throws IOException {
+				buffer.putChar((char) v);
+			}
+			
+			@Override
+			public void writeBytes(String s) throws IOException {
+				writeString(s);
+			}
+			
+			@Override
+			public void writeByte(int v) throws IOException {
+				buffer.put((byte) v);
+			}
+			
+			@Override
+			public void writeBoolean(boolean v) throws IOException {
+				writeBoolean(v);
+			}
+			
+			@Override
+			public void write(byte[] b, int off, int len) throws IOException {
+				throw new IOException("Not implemented");
+			}
+			
+			@Override
+			public void write(byte[] b) throws IOException {
+				buffer.put(b);
+			}
+			
+			@Override
+			public void write(int b) throws IOException {
+				buffer.put((byte) b);
+			}
+		};
+		
+		try {
+			CompressedStreamTools.write(compound, out);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return this;
+	}
```

Comment:
wtf??
so many better ways to do this for example wrap a Object/DataOutputStream around a ByteArrayOutputStream and write 100 lines less :)

Reply:
So uhh how does this work again?
```java
		DataOutputStream stream= new DataOutputStream(new ByteArrayOutputStream());
		
		try {
			CompressedStreamTools.write(compound, stream);
		} catch (IOException e) {
			e.printStackTrace();
		}
		buffer.put(stream);
```
This doesn't work but it was my first thought...